File: 백준/Silver/1978. 소수 찾기/소수 찾기.py
import math
input()
li = list(map(int, input().split()))
for x,y in enumerate(li):
    for i in range(2, int(math.sqrt(y))+1):
        if y%i == 0 or y<2:
            li[x]=0
            break
c=0
for i in li:
    if i>1:
        c=c+1
print(c)

File: 프로그래머스/lv2/42839. 소수 찾기/소수 찾기.js
function solution(numbers) {
    const set = new Set();
    numbers = numbers.split("");
    for(let i=0; i<numbers.length; i++){
        const perArr = [];
        const perCheck = Array(numbers.length).fill(false);
        perm(i);

        function perm(r){
            if(perArr.length == r+1){
                set.add(parseInt(perArr.join("")));
                return;
            }
            for(let j=0; j<numbers.length; j++){
                if(!perCheck[j]){
                    perArr.push(numbers[j]);
                    perCheck[j] = true;
                    perm(r);
                    perCheck[j] = false;
                    perArr.pop();
                }
            }
        }

    }
    return [...set].reduce((acc,curr) => acc+isPrime(curr), 0);
}
function isPrime(n){
    if(n<=1){
        return false;
    }
    for(let i=2; i<=Math.sqrt(n); i++){
        if(n%i==0) {
            return false;
        }
    }
    return true;
}